debian/tests/docker-in-lxd

```diff
@@ -165,22 +171,41 @@ lxc exec docker --env DEBIAN_FRONTEND=noninteractive -- apt-get full-upgrade -y
 lxc restart docker
 sleep 5
 
+# We call this again here because if the local apt repository lived
+# under /tmp, it won't be available anymore after a container restart.
+copy_local_apt_files
+lxc exec docker -- apt-get update
+
 lxc exec docker --env DEBIAN_FRONTEND=noninteractive -- apt-get install docker.io -y
 
 # Now basically run the simplest possible test inside the container.
 
 case ${arch} in
-    amd64)
-        image=hello-world;;
-    armhf|i386|s390x)
-        image=${arch}/hello-world;;
+    amd64|i386|s390x)
+        image_hello_world=${arch}/hello-world
+        image_ubuntu="${arch}/ubuntu:devel";;
+    armhf)
+        image_hello_world=arm32v7/hello-world
+        image_ubuntu=arm32v7/ubuntu:devel;;
     ppc64el)
-        image=ppc64le/hello-world;;
+        image_hello_world=ppc64le/hello-world
+        image_ubuntu=ppc64le/ubuntu:devel;;
     arm64)
-        image=aarch64/hello-world;;
+        image_hello_world=arm64v8/hello-world
+        image_ubuntu=arm64v8/ubuntu:devel;;
     *)
         echo "unknown arch: ${arch}"
         exit 2;;
 esac
 
-lxc exec docker -- docker run ${image}
+lxc exec docker -- docker run "${image_hello_world}"
+
+# We also need to run some commands inside an ubuntu:devel container,
+# to make sure that docker.io can work correctly with the latest
+# glibc.
+#
+# It's also important to make sure the docker container is using the
+# same apt configuration from the lxd container, which itself is the
+# same apt configuration from the host.
+lxc exec docker -- docker run -v /etc/apt:/etc/apt "${image_ubuntu}" \
```

Comment:
Now this is getting pretty inceptionish, but the lxd container's /etc/apt is for the same version as the host, whereas image_ubuntu is ubuntu:devel, right? So when this is backported to stable releases, it's going to fail because this is going to try to downgrade heaps of packages. We should probably only do this when /etc/lsb-release is the same between the container and the ubuntu:devel image or something?

Reply:
Argh, you're right.  The funny thing is that I thought about that problem at some point, but then forgot to revisit the code and fix it.  Oh, well...

Let me see what I can do to mitigate the problem.